tests/test_cli_compile.py

```diff
@@ -2457,6 +2447,27 @@ def test_triple_equal_pinned_dependency_is_used(
 )
 
 
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_not_specified_input_file(runner, make_module, fname, content):
+    """
+    It should raise an error if there are no input files or default input files
+    such as "setup.py", "requirements.in" or "pyproject.toml".
+    """
+    default_file_names = ("requirements.in", "setup.py", "pyproject.toml", "setup.cfg")
+    make_module(fname=fname, content=content, base_dir=os.getcwd())
+    out = runner.invoke(cli)
+    if fname in default_file_names:
```

Comment:
Usually, the tests should be simple and test one thing only. Once logic is added, they tend to become hard to read/understand as well as easy to be misleading/buggy.